Rubberduck.Inspections/Concrete/UnreachableCaseInspection/UCIValueResults.cs

```diff
@@ -0,0 +1,70 @@
+﻿using Antlr4.Runtime;
+using System;
+using System.Collections.Generic;
+
+namespace Rubberduck.Inspections.Concrete.UnreachableCaseInspection
+{
+    public interface IUCIValueResults
+    {
+        IUCIValue GetValue(ParserRuleContext context);
+        string GetTypeName(ParserRuleContext context);
+        string GetValueText(ParserRuleContext context);
+        bool Contains(ParserRuleContext context);
+        bool TryGetValue(ParserRuleContext context, out IUCIValue value);
+        void OnNewValueResult(object sender, ValueResultEventArgs e);
+        bool Any();
+    }
+
+    public class UCIValueResults : IUCIValueResults
+    {
+        private Dictionary<ParserRuleContext, IUCIValue> _parseTreeValues;
+
+        public UCIValueResults()
+        {
+            _parseTreeValues = new Dictionary<ParserRuleContext, IUCIValue>();
+        }
+
+        public IUCIValue GetValue(ParserRuleContext context)
+        {
+            if (context is null)
+            {
+                throw new ArgumentNullException();
+            }
+            return _parseTreeValues[context];
+        }
+
+        public string GetTypeName(ParserRuleContext context)
+        {
+            return GetValue(context).TypeName;
+        }
+
+        public string GetValueText(ParserRuleContext context)
+        {
+            return GetValue(context).ValueText;
+        }
+
+        public bool Contains(ParserRuleContext context)
+        {
+            return _parseTreeValues.ContainsKey(context);
+        }
+
+        public bool Any()
+        {
+            return _parseTreeValues.Count == 0;
```

Comment:
Wouldn't it be the negation? (return `true` when it's not ` == 0` ? Why not just do `return _parseTreeValues.Any();`?

Reply:
Yes - thanks.  Missed the logic error because the function is not used.  Removed it altogether.